### PR TITLE
Create coordset using computed extent

### PR DESCRIPTION
**Pull request by agcapps**
Create coordset using computed extent, after padding user's requested data extent.

Unlike other types of mesh, the strided_structured mesh has padding on the sides.  This fix makes the coordset to the size computed by adding any padding to the logical size requested by the user.

### CODE REVIEW

File: src/libs/blueprint/conduit_blueprint_mesh_examples.hpp [hunk 1]
```diff
@@ -72,14 +74,104 @@ namespace examples
...
+    /// one element in each dimension than arrays that store fields over
+    /// elements.  In strided structured meshes, the array dimensions of vertex
+    /// and element fields are decoupled.  In this example, vertex fields will
+    /// use a 7x6 array, not an 8x7 array as required by other Blueprint meshes.
+    /// Since there are 3x2 mesh elements, there will be 4x3 vertex field
+    /// values.  To match the element fields' padding as closely as possible,
+    /// the vertex field values are located at origin (2, 2).
+    ///
+    /// The diagram below shows the mesh.  Note that the origin is at the top
+    /// left.  For elements, a space indicates a padding element and a `d`
```
Comment (cyrush) [inline src/libs/blueprint/conduit_blueprint_mesh_examples.hpp]: Thought: 
The logical origin could be different with respect to how it is embedded in space. Not sure if saying top left will confuse folks?

Comment (agcapps) [reply src/libs/blueprint/conduit_blueprint_mesh_examples.hpp]: Perhaps rephrase to "Note that the diagram origin is at the top left."  What do you think?  I need something to anchor all the coordinates in the previous paragraph, not to mention orient the vertices that have no data (the blanks).

Comment (cyrush) [reply src/libs/blueprint/conduit_blueprint_mesh_examples.hpp]: That sounds good!

**Comment by cyrush** [comment]
For debugging and so we can verify things match expectations, I added a point topology that uses the underlying coordinates w/o striding.  Since the field values also match the cardinality of this coordset, we can also plot the fields on this topology as well. 

Here is an example:
Screen Shot 2022-12-19 at 11 59 07 AM


Here the meshes plot are our intended strided results, and the points are the original points.

**Comment by cyrush** [comment]
I want to make sure we have the correct logic for the `pad` cases.

Screen Shot 2022-12-19 at 12 11 13 PM
Screen Shot 2022-12-19 at 12 11 49 PM

**Comment by cyrush** [comment]
@agcapps  I think the out of the box case is good but I still have questions about pad case.
I expect we will have to iterate more both in conduit and VisIt. 
I am merging so we can get the current improvements in the next Conduit release.

**Pull request merged**